mantidproject/mantid
# Add automatic saving of ASCII files to Reflectometry GUI

- Adds autosave option to the Reflectometry GUI Save ASCII tab.
- Fixes broken enabling/disabling of the controls in the event tab.
- Tidies up the layout of the Save ASCII tab to make it clearer which options apply to autosave vs ad-hoc saves.
- Adds necessary hooks to data processor widget.

**To test:**
1. Open the ISIS Reflectometry Interface `Interfaces -> ISIS Reflectometry`
2. Navigate to the Save ASCII tab and set the save path to a directory which does not exist on your machine.
3. Attempt to enable automatic saving by checking the box.
4. Ensure you receive a warning message and the box remains unchecked.
5. Load the big table file from `\\olympic\babylon5\Public\GemmaGuest\issues\21674_disable_editing_on_refl_gui\INTER_NR_test3.tbl` using `Reflectometry->Import .TBL`.
6. Process the first group and ensure that you do not recieve any warnings.
7. Return to the SaveASCII tab, set the save path to a directory which does exists and check the box to enable autosave - this should succeed.
8. Return to the Runs tab and process the next group. This should produce a file named `IvsQ_13469_13470.dat`
10. Remove the second row from the third group and process the group. This should produce a file named `IvsQ_binned_13460.dat`
11. Experiment with the different file formats ensuring that processing groups produces a file with the correct format.
9. Change the save path back to a non-existent directory while autosave is enabled. You should receive and information pop up warning you that the directory does not exist.
10. Disable autosaving and process a different group. Ensure that a corresponding file is not produced.

Fixes #21691. 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
